### PR TITLE
Projection mo rtp

### PR DESCRIPTION
**Pull request by glb96**
Creates a print RTP routine to compute the projection of TD MO on reference one. 
The reference MOs can be from SCF calculation (by default the one from which the RTP starts) or XAS_TDP. 
A regtest has been added in the XAS_TDP folder in order to test the reading of .wfn obtained from XAS_TDP module as the reference MO to project on.

### CODE REVIEW

File: src/cp_control_types.F [hunk 1]
```diff
@@ -38,7 +38,24 @@ MODULE cp_control_types
       INTEGER, DIMENSION(2)                :: distribution_layout
       INTEGER                              :: blocked
    END TYPE pw_grid_option
+! **************************************************************************************************
+! \brief parameters for EMD/RTP calculations involving MO projections
+! **************************************************************************************************
+   TYPE proj_mo_type
```
Comment (oschuett) [inline src/cp_control_types.F:44]: Please add default initializes to this type: https://www.cp2k.org/conv#c016.

File: src/cp_control_utils.F [hunk 2]
```diff
@@ -2277,6 +2279,19 @@ SUBROUTINE read_rtp_section(dft_control, rtp_section)
       dft_control%rtp_control%delta_pulse_direction = tmp
       CALL section_vals_val_get(rtp_section, "SC_CHECK_START", &
                                 i_val=dft_control%rtp_control%sc_check_start)
+      tmp_section => section_vals_get_subs_vals(rtp_section, "PRINT%PROJECTION_MO")
```
Comment (oschuett) [inline src/cp_control_utils.F]: Maybe use a more useful variable name, like `proj_mo_section`.

File: src/emd/rt_projection_mo_utils.F [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+
+         ! Initialize the other parameters related to the TD mos.
+         CALL section_vals_val_get(proj_mo_section, "SUM_ON_ALL_REF", i_rep_section=i_rep, &
+                                   l_val=proj_mo%sum_on_all_ref)
+
+         CALL section_vals_val_get(proj_mo_section, "TD_MO_SPIN", i_rep_section=i_rep, &
+                                   i_val=proj_mo%td_mo_spin)
+         IF (proj_mo%td_mo_spin > SIZE(mos)) &
+            CALL cp_abort(__LOCATION__, &
+                          "You asked to project the time dependent BETA spin  while the "// &
```
Comment (oschuett) [inline src/emd/rt_projection_mo_utils.F]: Nit: Extra space between `spin  while`.

File: src/emd/rt_projection_mo_utils.F [hunk 4]
```diff
@@ -0,0 +1,571 @@
...
+                        file_status="OLD", &
+                        unit_number=restart_unit)
+      END IF
+
+      CALL read_mos_restart_low(mo_ref_temp, para_env=para_env, qs_kind_set=qs_kind_set, &
+                                particle_set=particle_set, natom=natom, &
+                                rst_unit=restart_unit)
+      !ALL read_mos_restart_low(mo_ref_temp, para_env=para_env, qs_kind_set=qs_kind_set, &
+      !                      particle_set=particle_set, natom=natom, &
+      !                      rst_unit=restart_unit, multiplicity=dft_control%multiplicity)
```
Comment (oschuett) [inline src/emd/rt_projection_mo_utils.F]: Left over?

File: src/emd/rt_projection_mo_utils.F [hunk 5]
```diff
@@ -0,0 +1,571 @@
...
+         ! fill with the reference mo values
+         CALL cp_fm_create(proj_mo%mo_ref(mo_index), mo_ref_fmstruct, 'mo_ref')
+         CALL cp_fm_to_fm(mo_ref_temp(proj_mo%ref_mo_spin)%mo_coeff, mo_coeff_temp, &
+                          ncol=1, &
+                          source_start=real_mo_index, &
+                          target_start=1)
+
+         ! multiply with overlap matrix to save time later on
+         CALL cp_dbcsr_sm_fm_multiply(matrix_s(1, 1)%matrix, mo_coeff_temp, proj_mo%mo_ref(mo_index), ncol=1)
+         !CALL cp_fm_to_fm(mo_coeff_temp, proj_mo%mo_ref(mo_index))
```
Comment (oschuett) [inline src/emd/rt_projection_mo_utils.F]: Left over?

**Review by oschuett — COMMENTED** [review]
Great work! Just a few minor comments.

I really like your well written error messages. They make such a difference for usability.

File: src/cp_control_types.F (hunk 1 above)
Comment (glb96) [reply src/cp_control_types.F:44]: I Have initialized the values, expect for the allocatable ones. Is it done properly?

File: src/cp_control_utils.F (hunk 2 above)
Comment (glb96) [reply src/cp_control_utils.F]: Should be fixed

File: src/emd/rt_projection_mo_utils.F (hunk 4 above)
Comment (glb96) [reply src/emd/rt_projection_mo_utils.F]: fixed

File: src/emd/rt_projection_mo_utils.F (hunk 5 above)
Comment (glb96) [reply src/emd/rt_projection_mo_utils.F]: fixed

**Comment by glb96** [comment]
Thanks for your help! I have pushed an updated version with your advice.

File: src/emd/rt_projection_mo_utils.F (hunk 3 above)
Comment (glb96) [reply src/emd/rt_projection_mo_utils.F]: oups! thx

File: src/cp_control_types.F (hunk 1 above)
Comment (oschuett) [reply src/cp_control_types.F:44]: Yes, it's perfect! Allocatables  are always automatically initialized

**Pull request merged**